diff --git a/rutabaga_gfx/Android.bp b/rutabaga_gfx/Android.bp
index 732920f13..f93a861a3 100644
--- a/rutabaga_gfx/Android.bp
+++ b/rutabaga_gfx/Android.bp
@@ -23,9 +23,7 @@ rust_library {
     edition: "2021",
     features: [
         "gfxstream",
-        "gfxstream_stub",
     ],
-    cfgs: ["fence_passing_option1"],
     rustlibs: [
         "libanyhow",
         "libcfg_if",
@@ -39,6 +37,36 @@ rust_library {
     ],
     proc_macros: ["libremain"],
     apex_available: ["com.android.virt"],
+    cfgs: [
+        "gfxstream_unstable",
+    ],
+    target: {
+        android: {
+            shared_libs: [
+                "libbase",
+                "libc++",
+                "libcutils",
+                "liblog",
+                "libnativewindow",
+            ],
+            static_libs: [
+                "libgfxstream_backend",
+            ],
+            rustlibs: [
+                "libnativewindow_bindgen",
+                "libnativewindow_rs",
+            ],
+        },
+        host: {
+            features: ["vulkano"],
+            shared_libs: [
+                "libgfxstream_backend",
+            ],
+            rustlibs: [
+                "libvulkano",
+            ],
+        },
+    },
 }
 
 rust_test {
@@ -57,9 +85,7 @@ rust_test {
     edition: "2021",
     features: [
         "gfxstream",
-        "gfxstream_stub",
     ],
-    cfgs: ["fence_passing_option1"],
     rustlibs: [
         "libanyhow",
         "libcfg_if",
@@ -73,6 +99,26 @@ rust_test {
         "libzerocopy",
     ],
     proc_macros: ["libremain"],
+    static_libs: [
+        "libgfxstream_backend",
+    ],
+    shared_libs: [
+        "libc++",
+        "libbase",
+    ],
+    target: {
+        android: {
+            shared_libs: [
+                "libnativewindow",
+                "libcutils",
+                "liblog",
+            ],
+            rustlibs: [
+                "libnativewindow_bindgen",
+                "libnativewindow_rs",
+            ],
+        },
+    },
 }
 
 // Manually written variant of librutabaga_gfx.
@@ -121,6 +167,10 @@ rust_library {
                 "fence_passing_option1",
                 "gfxstream_unstable",
             ],
+            rustlibs: [
+                "libnativewindow_bindgen",
+                "libnativewindow_rs",
+            ],
         },
     },
     apex_available: ["com.android.virt"],
